### ISSUE TITLE
New Middleware Proposal: validator-standalone

### ISSUE THREAD

**Issue by nponeccop** [middyjs/middy#639]
Proposal for a lighter version of `middy-validator` which uses https://ajv.js.org/standalone.html instead of "normal" ajv.

The current `middy-validator` is designed to work with "normal" ajv, which is a big package, takes a lot of space in the deployment S3 bucket and introduces ~200 ms cold start latency in `serverless-offline` on my relatively slow system. 

I'm not sure of real impact on the AWS billing though, so the performance consideration should be secondary. The primary consideration is sticking to the spirit of `connect`: tiny middleware, as unopinionated as possible, performing only one function. 

I mean, not everyone needs bells and whistles of internationalization, good messages and whatnot. The standalone code generated by `ajv` is really tiny, it's identical to what most people would write by hand, at least in my case of a small and simple schema.

Look at my simple Serverless service, doing work of some 5 serverless middlewares including `validate`, all in 2.8 kB deployment size reported in the AWS Console:

https://github.com/nponeccop/serverless-openapi-templates/blob/fa715f90940ea3bf189d1d265b0304981eb88517/validation/handler.js

`schema/helloPost.js` is generated by running `node openapi-schemas.js` manually.

I'd like to rewrite the app so it is based on `middy` components. For that I need to a version of `validator` which doesn't depend on `ajv`, or at least doesn't require it if it isn't asked.

Note that I chose only needed JS manually, and my code does much less than the middy components do. But I think you understand the idea to make middlewares as small as possible.

**Comment by willfarrell** [comment]
Yeah, I thought quite a bit on this while writing v2. Backwards compatibility, total size, performance and inclusivity were all considered. Because v2 had some big breaking changes I opted for backwards compatibility and inclusivity for now. Making this more performant while staying dev friendly is a goal for v3.

During testing I ran into what you've described, which led to the addition of `Important Compiling schemas on the fly will cause a 50-100ms performance hit during cold start for simple JSON Schemas. Precompiling is highly recommended.` in the docs. Keep in mind ajv v7 was just released right before the alpha went out and we updated to ajv v8 last week. Since it's gotten funding it's been moving quickly. Updating `i18n` is up next I'm told. 

I'm open to solutions to solving this. Currently, it feels like a script of some kind is needed (maybe a rollup template?). Let me know what you land on.
